web/components/rsvp/index.js

```diff
@@ -0,0 +1,213 @@
+
+import praise from '../../assets/emoji/praise.svg';
+import sparkle from '../../assets/emoji/sparkle.svg';
+import celebrate from '../../assets/emoji/celebrate.svg';
+
+import rsvpSuccessGraphic from '../../assets/rsvpSuccess.svg';
+
+class RSVP extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      isIDValid: true,
+      isSubmitted: false,
+      isSubmitting: false,
+      submitFailed: false,
+      emergencyContactName: '',
+      emergencyContactNumber: '',
+      tShirtSize: '',
+      dietRestriction: '',
+      wouldLikeToSee: '',
+    };
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    const { params } = match;
+    const { id } = params;
+    const { firebase } = this.props;
+
+    Axios.post(
+      firebase.nwUtils.getFunctionUrl('checkValidID'),
+      { id },
+      { headers: { 'Content-Type': 'text/plain' } }
+    ).then((res) => {
+      if (res.status === 200) {
+        const { accepted, rsvped, firstName } = res.data;
+
+        if (!accepted) {
+          console.log('Did not find valid ID');
+          this.setState({ isIDValid: false });
+        } else if (rsvped) {
+          console.log('already RSVPed');
+          this.setState({ isSubmitted: true });
+        } else {
+          console.log('Found ID: ' + firstName);
+          // this.setState({ isIDValid: true });
+          this.setState({ firstName });
+        }
+      }
+    });
+  }
+
+  render() {
+    const { isIDValid, isSubmitted, isSubmitting, submitFailed } = this.state;
+
+    if (!isIDValid) return (<Redirect to="/page_not_found" />);
+    if (isSubmitted) {
+      return (
+        <div className="pad-nav fill-min-height flex dir-col ai-center jc-center pad-sides-l">
+          <img alt="Success!" className="pad-ends-s" max-height="25%" src={rsvpSuccessGraphic} />
+          <br />
+          <h2 className="text-align-center">Thanks for confirming your attendance! <img className="emoji" alt="✨" src={sparkle} /></h2>
+          <p className="text-align-center" style={{ maxWidth: 680 }}>You’ve successfully secured your spot.
+            We’ll see you bright and early on January 26th. We’re so excited to have you at nwHacks!
+          </p>
+          <br />
+          <Link to="/" className="pad-bottom-l">
+            <SecondaryButton text="<- Take me back home" />
+          </Link>
+        </div>
+      );
+    }
+
+    const { firebase } = this.props;
+    const { match } = this.props;
+    const { params } = match;
+    const { id } = params;
+    const {
+      firstName,
+      emergencyContactName,
+      emergencyContactNumber,
+      tShirtSize,
+      dietRestriction,
+      wouldLikeToSee,
+    } = this.state;
+
+    return (
+      <div className="pad-nav application fill-width flex jc-center">
+        <div className="pad-top-mega pad-bottom-tera">
+          <h2>Hi {firstName}, please RSVP <img className="emoji" alt="🙌" src={praise} /></h2>
```

Comment:
Minor detail, when the `firstName` is still loading, this renders as `Hi , please RSVP` with a space before the comma. We can fix that by adding that space + `firstName` iff `firstName` does exist.

Reply:
Fixed!